src/jit/gentree.cpp

```diff
@@ -11931,11 +11967,11 @@ GenTree* Compiler::gtFoldTypeCompare(GenTree* tree)
         GenTree* compare = nullptr;
         if ((op1TunneledHandle != nullptr) && (op2TunneledHandle != nullptr))
         {
-            compare = gtNewOperNode(oper, TYP_INT, op1TunneledHandle, op2TunneledHandle);
+            compare = gtCreateHandleCompare(oper, op1TunneledHandle, op2TunneledHandle, !IsTargetAbi(CORINFO_CORERT_ABI));
```

Comment:
It is possible to just compare the pointers for number of cases in CoreRT as well. It would better for this to be answer by the JIT/EE interface and not hard coded to CoreRT ABI.

Reply:
Can we get away with the existing JitInterface method (`canInlineTypeCheckWithObjectVTable`), or do we need a new one? (I know on CoreRT the answer for both "can I compare two typehandles?" and "can I compare vtable with a typehandle" is the same, but does that apply to CoreCLR too?)

Reply:
That answer is not same for CoreCLR: You can always compare two typehandles, but you cannot always compare vtable with a typehandle in CoreCLR today.

Reply:
Sigh. So a new JitInterface method it is. I really wish there wasn't so much ceremony around adding those :(.